### ISSUE TITLE
my cuckoo is showing  every file/url is malicious 

### ISSUE THREAD

**Issue by ESL482** [cuckoosandbox/cuckoo#2517]
Thanks for creating an issue! But first: did you read our community guidelines?
https://cuckoo.sh/docs/introduction/community.html

##### My issue is:
After analysis cuckoo is not giving the correct results.Even when i entered url as "cuckoosandbox.org" also it showing 3.8  score.After white listing domains in domain.txt also it is showing many genuine  urls as malicious.Please help me to know how to whitelist urls also.
![some urls](https://user-images.githubusercontent.com/42828058/46660317-630b7280-cbd4-11e8-9493-e3c098624ed2.png)
 also.
i have doubt that whether installing cuckoo community is enough or , do i need to install yara-master separately and have to integrate with cuckoo.
![cuckoo](https://user-images.githubusercontent.com/42828058/46660176-1aec5000-cbd4-11e8-8355-78d19f53eb24.png)
 

##### My Cuckoo version and operating system are: cuckoo 2.0.6 , host: ubuntu 16.04 , guest:windows 7

due to this signatures issue every file is showing malicious , so i am not able to conclude which is malicious also, please help me to get out of it

**Comment by doomedraven** [comment]
don't do yara master if you referer to yara rules it will breaks your cuckoo, 
the problem here is what it grabs all urls, as it expect to analyze malware only! so it can't do difference,  my suggestion is to whitelist the urls which you don't want to popup there and done

the urls are in browsers as default page etc, so that is the problem, they are in process memory

**Comment by ESL482** [comment]
in domains.txt ,i am whitelisted some domains even though that domains are repeating again and how to resolve dead host problem ,like it is connecting to my vm ip with different ports and it giving like deadhost and one of signature like this " 

request | POST http://127.0.0.1:8000http://127.0.0.1:8000/status
" also

**Comment by doomedraven** [comment]
what about whipe browser memory/history?

**Comment by ESL482** [comment]
i cleared all cache, cookies, and history in vm and i enabled dns_whitelist in processing.conf also

**Comment by doomedraven** [comment]
well something is bad with configuration for sure, and about the dead host of your host, why just not to start wireshark and see which services do that noise?

+that urls should be whitelisted in signature than in dns_whitelist, or both :)

**Comment by ESL482** [comment]
may i know how to whitelist in signature ?

**Comment by doomedraven** [comment]
you have signature name `http traffic contains....` go here https://github.com/cuckoosandbox/community and do search by that :) https://github.com/cuckoosandbox/community/search?q=http+traffic+contains&unscoped_q=http+traffic+contains

tadan https://github.com/cuckoosandbox/community/blob/b26204351f10aff78cc8528a31dc35b346a80424/modules/signatures/network/network_cnc_http.py :)

**Comment by ESL482** [comment]
thanks fo this, do i need to edit below right? and what i need to do to reslove that deadhost  issue
![whitelis](https://user-images.githubusercontent.com/42828058/46662179-5c7efa00-cbd8-11e8-877f-96254d8c81f4.png)

**Comment by doomedraven** [comment]
yes you can edit it, well for dead host, you can whitelist your host also there, but perfect solution is to see which service in windows generate that request and stop it, use process explorer, process hacker, etc any tool which you need

**Comment by ESL482** [comment]
ok...thanks...And one more thing also , i am using manual proxy in my vm browser for that i need to give credentials nad i set firefox as my default browser and  i gave settings in like remember my credentilas, but every time cuckoo is going to open IE only, can any solution for that like any editing in conf files

**Comment by doomedraven** [comment]
edit packages, any url is set to be opened in IE, so it doesn't care about defautl browser https://github.com/cuckoosandbox/cuckoo/blob/3b628384d5dd2c8982e71548d7413e2c89172181/cuckoo/core/submit.py#L172

**Comment by ESL482** [comment]
sorry i am not getting ,you are saying that i have to give give firefox url instead of "url" above.i am not getting how to edit this package.
![url](https://user-images.githubusercontent.com/42828058/46662908-483bfc80-cbda-11e8-9d52-83c0714b0331.png)

**Comment by doomedraven** [comment]
do you see the `package":"ie"` that mean if its url run with internet explorer not firefox

**Comment by ESL482** [comment]
can't we give "mozilla" in place of "ie"

**Comment by doomedraven** [comment]
you can do what ever you want, you have code on your host, its opensource ;)

**Comment by ESL482** [comment]
ok..thanks..i will check remaining and revert back

**Comment by ESL482** [comment]
Hi , i set firefox as package and its working fine , i have another doubt while writing my own signature

from cuckoo.common.abstracts import Signature

class CreatesExe(Signature):
    name = "internal_mail"
    description = "Awareness mail"
    severity = 1
    categories = ["generic"]
    authors = ["ESL"]
    minimum = "2.0"

    def on_complete(self):
        return self.check_file(pattern=".*\\.exe$", regex=True)

// Actually i want to set an url in the pattern like , pattern="http://0.0.0.0:port?rid=*"
in return what have to give to execute this signature and i am writing this signature under network folder, please rectify above signature if any mistake.

**Comment by ESL482** [comment]
hi doomedraven , when i searched about abstracts.py , i found many files.can you share any link to understand the process about abstracts.py and how to set my own url signature
i already follwed the below link
https://cuckoo.sh/docs/customization/signatures.html?highlight=signatures

**Comment by doomedraven** [comment]
here is another trick `from cuckoo.common.abstracts import Signature` <- see from cuckoo.common.abstracts, you can walk folders in github cuckoo/common/abstracts.py ;)

https://github.com/cuckoosandbox/cuckoo/blob/master/cuckoo/common/abstracts.py

**Comment by ESL482** [comment]
from the example of bot_solar_url.py , i understood that i have to give my signature like below
 **urls_re = [
        "192.168.56.101:8080?rid=*",
    ]

    def on_complete(self):
        for indicator in self.urls_re:
            match = self.check_url(pattern=indicator, regex=True)
            if match:
                self.mark_ioc("url", match)
                return True**
is this is right ? i have a doubt that's why i want to be clear before setting my own signature

**Comment by doomedraven** [comment]
according to documentation yes ;)

```
   def check_url(self, pattern, regex=False, all=False):
        """Checks for a URL being contacted.
        @param pattern: string or expression to check for.
        @param regex: boolean representing if the pattern is a regular
                      expression or not and therefore should be compiled.
        @return: boolean with the result of the check.
        """
        urls = set()
        for item in self.get_net_http():
            urls.add(item["uri"])

        return self._check_value(pattern=pattern,
                                 subject=list(urls),
                                 regex=regex,
                                 all=all)
```

**Comment by ESL482** [comment]
Hi ,
I put that signature in cuckoo/signatures/windows
at the time of analysis it showing
2018-10-16 11:50:29,910 [cuckoo.processing.behavior] WARNING: Analysis results folder does not contain any behavior log files.
and the it is not matching with signature , i didn't get any description like i gave in signature in report.
my signature is 

from lib.cuckoo.common.abstracts import Signature

class InternalURL(Signature):
    name = "internal_url"
    description = "This mail was part of an internal awareness."
    severity = 2
    categories = ["internal"]
    authors = ["ESL"]
    minimum = "2.0"
    urls_re = [
        **"http://192.168.56.101:7070?rid=*",**
       
    ]

    def on_complete(self):
        for indicator in self.urls_re:
            match = self.check_url(pattern=indicator, regex=True)
            if match:
                self.data.append({"url": match})
                return True
, do i need to change any thing else in abstarcts.py , why my signature is not matching with that , when i gave 
**http://192.168.56.101:7070?rid=V1i701w**

**Comment by ESL482** [comment]
i tried by givivng category as "generic" also,

**Comment by doomedraven** [comment]
for start lets go by parts
1. do you see what cuckoo running your signature? (run in debug mode `-d`, `cuckoo -d process -r id`)
2. do you see that url in report?
3. did you try to do regex in python console to make sure it matchs?

**Comment by ESL482** [comment]
1. I am already running in debug mode.( i attached the logs)
2.no, in report my url is not coming.
![console](https://user-images.githubusercontent.com/42828058/47133398-4226d980-d2c6-11e8-9316-027b1c1fa276.png)
[cuckoo.log](https://github.com/cuckoosandbox/cuckoo/files/2490359/cuckoo.log)

**Comment by doomedraven** [comment]
Signatures cant be executed like this

**Comment by ESL482** [comment]
then , may i know how to do that .

**Comment by doomedraven** [comment]
Cuckoo execute it when you doing analysis or cuckoo -d process -r id to reprocess

Documentation read it ;)

**Comment by ESL482** [comment]
when i run using process id , mine is showing in signatures field , find the attached logs for information .
But my signature is not matching with given url, please give an idea how to resolve this.
[logs.txt](https://github.co
[Matched signatures.log](https://github.com/cuckoosandbox/cuckoo/files/2490571/Matched.signatures.log)
m/cuckoosandbox/cuckoo/files/2490569/logs.txt)

**Comment by doomedraven** [comment]
add logging to your signature or use ide with debuger to debug it

**Comment by ESL482** [comment]
[console.log](https://github.com/cuckoosandbox/cuckoo/files/2490578/console.log)
[cuckoo.log](https://github.com/cuckoosandbox/cuckoo/files/2490580/cuckoo.log)

**Comment by kyrozhyper** [comment]
I want to ask, how can scores and signatures be displayed?
because in my place I didn't appear
